multiversx_sdk_cli/cli_delegation.py

```diff
@@ -369,66 +390,233 @@ def ensure_arguments_are_provided_and_prepared(args: Any):
+    guardian_address = cli_shared.get_guardian_address(guardian, args)
+    return guardian, guardian_address
+
+
+def prepare_relayer(args: Any):
+    relayer = cli_shared.load_relayer_account(args)
+    relayer_address = cli_shared.get_relayer_address(relayer, args)
+    return relayer, relayer_address
+
+
 def do_create_delegation_contract(args: Any):
-    ensure_arguments_are_provided_and_prepared(args)
+    validate_transaction_args(args)
+    ensure_wallet_args_are_provided(args)
+    validate_broadcast_args(args)
+    validate_chain_id_args(args)
 
-    sender = cli_shared.prepare_account(args)
-    config = TransactionsFactoryConfig(args.chain)
+    sender, nonce = prepare_sender(args)
+    guardian, guardian_address = prepare_guardian(args)
+    relayer, relayer_address = prepare_relayer(args)
+
+    chain_id = cli_shared.get_chain_id(args.chain, args.proxy)
+    config = TransactionsFactoryConfig(chain_id)
     delegation = DelegationOperations(config)
 
-    tx = delegation.prepare_transaction_for_new_delegation_contract(sender, args)
+    gas_limit = args.gas_limit if args.gas_limit else 0
+
+    tx = delegation.prepare_transaction_for_new_delegation_contract(
+        owner=sender,
+        native_amount=int(args.value),
+        total_delegation_cap=int(args.total_delegation_cap),
+        service_fee=int(args.service_fee),
+        gas_limit=gas_limit,
+        gas_price=int(args.gas_price),
+        nonce=nonce,
+        version=int(args.version),
+        options=int(args.options),
+        guardian_account=guardian,
+        guardian_address=guardian_address,
+        relayer_account=relayer,
+        relayer_address=relayer_address,
+        guardian_service_url=args.guardian_service_url,
+        guardian_2fa_code=args.guardian_2fa_code,
+    )
+
     cli_shared.send_or_simulate(tx, args)
 
 
 def get_contract_address_by_deploy_tx_hash(args: Any):
-    args = utils.as_object(args)
+    validate_proxy_argument(args)
 
     config = get_config_for_network_providers()
     proxy = ProxyNetworkProvider(url=args.proxy, config=config)
-
     transaction = proxy.get_transaction(args.create_tx_hash)
-    transaction_events = transaction.logs.events
-    if len(transaction_events) == 1:
-        contract_address = transaction_events[0].address
-        print(contract_address.to_bech32())
-    else:
-        raise errors.ProgrammingError(
-            "Tx has more than one event. Make sure it's a staking provider SC Deploy transaction."
-        )
+
+    parser = DelegationTransactionsOutcomeParser()
+    outcome = parser.parse_create_new_delegation_contract(transaction)
+
+    if len(outcome) > 1:
+        print("This transaction created more than one delegation contract.")
+
+    for i in range(len(outcome)):
+        print(f"Delegation contract address: {outcome[i].contract_address.to_bech32()}")
 
 
 def add_new_nodes(args: Any):
-    ensure_arguments_are_provided_and_prepared(args)
+    validate_transaction_args(args)
```

Comment:
can you extract this into a method because they are used in every method

Reply:
done